File: Python/S0011.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        res = 0
        while l < r:
            long = r - l
            current = long * min(height[l], height[r])
            if current > res: res = current
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return res

File: C++/S0011.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0, r = heights.size() - 1;
        int res = 0;
        while (l < r) {
            int lengths = r - l;
            int cur_water = lengths * min(heights[l], heights[r]);
            res = max(res, cur_water);
            if (heights[l] < heights[r]) l++;
            else r--;
        }
        return res;
    }
};